website/server/controllers/api-v3/groups.js

```diff
@@ -944,6 +940,12 @@ api.removeGroupMember = {
       member.save(),
       group.save(),
     ]);
+
+    if (group.hasNotCancelled())  {
```

Comment:
Is it worth making this `if (isInGroup & group.hasNotCancelled())  {` so that we don't run the check if the user was merely invited? That's slightly less load on the database (not really an issue I guess) and it also protects the user if `cancelGroupSubscriptionForUser()` develops any bugs in future that cause it to cancel subscriptions from group plans inappropriately (perhaps that's just paranoia).

Reply:
HasNotCancelled does call the DB. But for this route we are assuming they are in a group. So isInGroup wouldn't prevent that. Or it would always be false because we are cancelling after removing them now.

Reply:
`isInGroup` is just a variable ("guild" / "party", and effectively a binary flag when used like this) and it isn't changed after they've been removed so it shouldn't be false if it started as true at the beginning of the function.
The function is also run if `isInvited` is true (i.e., to remove an invitation to a group, rather than removing a group member) and that's the case where it's not necessary to attempt to remove a free subscription.

Reply:
I don't think that HasNotCancelled calls the db, it only uses local data

Reply:
Oh I didn't realize we use this same route for invitations. Then yes, you are correct. I'll ad that check in